src/main/java/org/codehaus/groovy/control/DefaultImportedClassCollectorHelper.java

```diff
@@ -0,0 +1,226 @@
+        }
+
+        public String getJavaVersion() {
+            return javaVersion;
+        }
+
+        public void setJavaVersion(String javaVersion) {
+            this.javaVersion = javaVersion;
+        }
+
+        public String getGroovyVersion() {
+            return groovyVersion;
+        }
+
+        public void setGroovyVersion(String groovyVersion) {
+            this.groovyVersion = groovyVersion;
+        }
+
+        public String[] getPackageNames() {
+            return packageNames;
+        }
+
+        public void setPackageNames(String[] packageNames) {
+            this.packageNames = packageNames;
+        }
+
+        public Map<String, Set<String>> getClassNameToPackageMap() {
+            return classNameToPackageMap;
+        }
+
+        public void setClassNameToPackageMap(Map<String, Set<String>> classNameToPackageMap) {
+            this.classNameToPackageMap = classNameToPackageMap;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (!(o instanceof DefaultImportedClassesInfo)) return false;
+            DefaultImportedClassesInfo that = (DefaultImportedClassesInfo) o;
+            return Objects.equals(javaVersion, that.javaVersion) &&
+                    Objects.equals(groovyVersion, that.groovyVersion) &&
+                    Arrays.equals(packageNames, that.packageNames);
+        }
+
+        @Override
+        public int hashCode() {
+            int result = Objects.hash(javaVersion, groovyVersion);
+            result = 31 * result + Arrays.hashCode(packageNames);
+            return result;
+        }
+
+        @Override
+        public String toString() {
+            return "DefaultImportedClassesInfo{" +
+                    "javaVersion='" + javaVersion + '\'' +
+                    ", groovyVersion='" + groovyVersion + '\'' +
+                    ", packageNames=" + Arrays.toString(packageNames) +
+                    ", classNameToPackageMap=" + classNameToPackageMap +
+                    '}';
+        }
+    }
+
+    private static class LazyInfoHolder {
+        private static final String SNAPSHOT = "SNAPSHOT";
+        private static final File CLASSES_INFO_FILE = new File(new File(GroovySystem.getGroovyRoot(), "caches"), "default_imported_classes");
+        private static final String JAVA_VERSION = System.getProperty("java.version");
+        private static final String GROOVY_VERSION = getGroovyVersion();
+
+        private static String getGroovyVersion() {
+            String version = GroovySystem.getVersion();
+            version = (null == version || version.trim().isEmpty()) ? SNAPSHOT : version;
+
+            // if using SNAPSHOT version(e.g. 3.0.0-SNAPSHOT), cache for only 1 hour because default imported classes may change when developing
+            return version.contains(SNAPSHOT) ? version + new SimpleDateFormat("-yyyyMMdd.HH").format(new Date()) : version;
```

Comment:
:x: I think this is problematic. We shouldn't bake in a rule like this. Ideally this should be a hash of the Groovy runtime.

Reply:
"Ideally this should be a hash of the Groovy runtime." -- could you provide me more details about Groovy runtime?

Reply:
@melix  "I would also avoid adding another third party dependency just for this: this is core groovy." -- Yep, I agree with you. so the third party dependency is optional, as you can see, the dependency is `compileOnly`